presto-main/src/main/java/com/facebook/presto/execution/buffer/PartitionedOutputBuffer.java

```diff
@@ -52,6 +57,10 @@
     private final AtomicLong totalPagesAdded = new AtomicLong();
     private final AtomicLong totalRowsAdded = new AtomicLong();
 
+    private final ConcurrentMap<Lifespan, AtomicLong> outstandingPagePerLifespan = new ConcurrentHashMap<>();
+    private final Set<Lifespan> noMorePagesForLifespan = ConcurrentHashMap.newKeySet();
+    private Consumer<Lifespan> lifespanFinishedCallback = null;
+
```

Comment:
Looks like my comments above for `ArbitraryOutputBuffer.java` applies here as well.